.github/workflows/test.yml

```diff
@@ -12,8 +12,6 @@ jobs:
     steps:
     - name: Install build tools
       run: sudo apt-get install -y build-essential
-    - name: Remove libssl-dev
-      run: sudo apt-get remove -y libssl-dev
```

Comment:
Does this mean the tests depend on the ubuntu-provided libssl-dev being installed first, then installing over it with our own build of 3.0? Or are we using both (openssl from ubuntu, but FIPS provider from our own build)?

Reply:
I removed this because we don't need development headers anymore, which was why we were installing libssl-dev.

Reply:
I must be missing something... it looks to me like this PR is removing the removal of libssl-dev, so after this PR we'll actually be leaving it installed after `install build-essential` pulls it in.

Reply:
My bad. We were previously removing the system openssl headers because we needed to use our own, so we were removing the system openssl headers to avoid conflicts. Now we don't care if there are headers or not, so there is no point on removing them.

Reply:
Got it! Thanks.